File: 011 Container With Most Water/ContainerWithMostWater.java
class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int n = height.length;
        int lo = 0;
        int hi = n - 1;
        int maxArea = 0;
        while (lo < hi) {
            maxArea = Math.max(maxArea, Math.min(height[lo], height[hi]) * (hi - lo));
            if (height[lo] <= height[hi]) {
                lo++;
            }
            else {
                hi--;
            }
        }
        return maxArea;
    }
}

File: 11 Container With Most Water/container_with_most_water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height);
        lo = 0
        hi = n - 1 
        max_area = 0
        while lo < hi:
            max_area = max(max_area, min(height[lo], height[hi]) * (hi - lo))
            if height[lo] <= height[hi]:
                lo += 1
            else:
                hi -= 1
        return max_area
        